### ISSUE TITLE
synchronize 'pull' mode incorrectly prepends PWD to src

### ISSUE THREAD

**Issue by acthp** [ansible/ansible#17492]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

plugins/action/synchronize.py
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides

```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

'pull' mode preprends PWD to the (remote) src path, e.g.

/home/blah/blah/rsync://some-host/path/on/remote
##### STEPS TO REPRODUCE

Try the example in the synchronize docs:

```
synchronize: mode=pull src=rsync://somehost.com/path/ dest=/some/absolute/path/

```

##### EXPECTED RESULTS

rsync from somehost.com/path/
##### ACTUAL RESULTS

failed rsync from ${PWD}rsync://somehost.com/path/

```

```

The problem seems to be around the dest_is_local flag. If dest_is_local (true for pull with remote src), the 'rsync://' prefix is not handled.

**Comment by bwheatley** [comment]
Can confirm the same issue for me. In Ansible {{1.9.4}}

**Comment by tima** [comment]
This should have been fixed with v2.2 by these two PRs. 
- https://github.com/ansible/ansible/pull/16756
- https://github.com/ansible/ansible-modules-core/pull/4211

Please try it out to verify.

**Comment by tima** [comment]
needs_info

**Comment by ansibot** [comment]
@acthp ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by whitequark** [comment]
This is still an issue as of Ansible 2.4.

**Comment by ansibot** [comment]
@acthp This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@acthp You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
